core/src/test/java/ai/timefold/solver/core/preview/api/variable/declarative/concurrent/ConcurrentValuesShadowVariableTest.java

```diff
@@ -112,12 +113,12 @@ void groupChain() {
         var entity2 = new TestdataConcurrentEntity("v2");
         var entity3 = new TestdataConcurrentEntity("v3");
 
-        var valueA1 = new TestdataConcurrentValue("a1");
-        var valueA2 = new TestdataConcurrentValue("a2");
-        var valueB1 = new TestdataConcurrentValue("b1");
-        var valueB2 = new TestdataConcurrentValue("b2");
-        var valueB3 = new TestdataConcurrentValue("b3");
-        var valueC = new TestdataConcurrentValue("c");
+        var valueA1 = Mockito.spy(new TestdataConcurrentValue("a1"));
+        var valueA2 = Mockito.spy(new TestdataConcurrentValue("a2"));
+        var valueB1 = Mockito.spy(new TestdataConcurrentValue("b1"));
+        var valueB2 = Mockito.spy(new TestdataConcurrentValue("b2"));
+        var valueB3 = Mockito.spy(new TestdataConcurrentValue("b3"));
+        var valueC = Mockito.spy(new TestdataConcurrentValue("c"));
 
         var concurrentGroupA = List.of(valueA1, valueA2);
         var concurrentGroupB = List.of(valueB1, valueB2, valueB3);
```

Comment:
How were the values for the groups selected? Based on what criteria? How can I verify that the grouping is correct?

I see values of the same type, with the same sources. There is nothing which gives me any indication that these should all not be grouped together, or that they should be grouped at all.